components/builder-api/src/server/resources/pkgs.rs

```diff
@@ -420,6 +421,90 @@ async fn download_package(req: HttpRequest,
                        &conn)
     {
         Ok(package) => {
+            let channels = match channels_for_package_ident(&req, &package.ident, target, &conn) {
+                Ok(channels) => channels,
+                Err(err) => {
+                    return HttpResponse::InternalServerError()
+                        .body(format!("Failed to determine package channels: {}", err));
+                }
+            };
+
+            let should_restrict = if let Some(chs) = channels.as_ref() {
+                let chs_lower: Vec<String> = chs.iter().map(|c| c.to_lowercase()).collect();
+
+                let in_unrestricted_channels = state.config
+                                                    .api
+                                                    .unrestricted_channels
+                                                    .iter()
+                                                    .any(|c| chs_lower.contains(&c.to_lowercase()));
+
+                let in_partially_unrestricted_channels =
+                    state.config
+                         .api
+                         .partially_unrestricted_channels
+                         .iter()
+                         .any(|c| chs_lower.contains(&c.to_lowercase()));
+
+                let in_restricted_if_present = state.config
+                                                    .api
+                                                    .restricted_if_present
+                                                    .iter()
+                                                    .any(|c| chs_lower.contains(&c.to_lowercase()));
+
+                !(in_unrestricted_channels
+                  || (in_partially_unrestricted_channels && !in_restricted_if_present))
+            } else {
+                true
+            };
+
+            if should_restrict {
```

Comment:
We only want to restrict on SAAS builder and not private on-prem builders. If `restricted_if_present` is empty, lets default to everything being unrestricted.